activerecord/lib/active_record/serialization.rb

```diff
@@ -10,6 +10,12 @@ module Serialization
       self.include_root_in_json = false
     end
 
+    module ClassMethods
+      def as_json(*)
+        name
```

Comment:
Are we assuming that we want the `.as_json` for a class name to only output the name of the class? Not even with brackets? Look at this an your tests, it doesn't seem like we're delivering json. Furthermore, is this overriding the normal behavior of when I call `.as_json` on a class before tables are loaded.

I think the focus here is that we want to solve why we're getting a stack overflow error. This seems more like handling it in a bit of a strange way.